src/NHSD.BuyingCatalogue.Documents.API/Controllers/SolutionsController.cs

```diff
@@ -12,16 +12,16 @@ namespace NHSD.BuyingCatalogue.Documents.API.Controllers
     [AllowAnonymous]
     public class SolutionsController : ControllerBase
     {
-        private readonly IStorage _storage;
+        private readonly IDocumentRepository _documentRepository;
 
-        public SolutionsController(IStorage storage)
-            => _storage = storage;
+        public SolutionsController(IDocumentRepository documentRepository)
+            => _documentRepository = documentRepository;
 
         [HttpGet]
         [Route("{id}/documents/")]
         [ProducesResponseType(typeof(IAsyncEnumerable<string>), (int)HttpStatusCode.OK)]
```

Comment:
If we're sticking with the ActionResult<T> return type then we shouldn't need to specify the type in the `ProducesResponseTypeAttribute` constructor.

We should also use `StatusCodes` instead of `HttpStatusCodes` as it will avoid the need to cast.

For example,

```c#
[ProducesResponseType(typeof(IAsyncEnumerable<string>), (int)HttpStatusCode.OK)]
```

becomes

```c#
[ProducesResponseType(StatusCodes.Status200OK)]
```

The main argument against using ActionResult<T> would be the negative impact it has on test readability, in which case we would need to continue to pass the type to `ProducesResponseTypeAttribute`.